Add spring tests(ShoppingCart entity):

DEz-1996 opened the pull request:
Add tests for ShoppingCart entity:
1. ShoppingCartServiceImplTest.java
2. ShoppingCartRepositoryTest.java
3. ShoppingCartControllerTest.java
Currnt covarage:
![coverage](https://github.com/DEz-1996/online-book-store/assets/127430483/b01cd5f5-9582-453a-989e-3f25164bbd5b)

DEz-1996 committed:
Add docker:
1. Add .env file
2. Add Dockerfile
3. Add docker compose dependency
4. Add docker-compose.yml with mysql db image

DEz-1996 committed:
Add spring tests:
Add tests for book entity:
1. BookServiceTest.java
2. BookRepositoryTest.java
3. BookControllerTest.java

DEz-1996 committed:
Add spring tests:
Add tests for book entity:
1. CustomMySqlContainer.java
2. BookServiceTest.java
3. BookRepositoryTest.java
4. BookControllerTest.java

DEz-1996 committed:
Add spring tests:
Add tests for ShoppingCart entity:
1. ShoppingCartServiceImplTest.java
2. ShoppingCartRepositoryTest.java
3. ShoppingCartControllerTest.java

TarasYurkovskyi reviewed (approved):
Good job!

yevheniisynytsia commented on src/test/java/online/book/store/controller/ShoppingCartControllerTest.java:
```diff
@@ -0,0 +1,103 @@
+            "classpath:database/cart_items/insert-cart-items.sql",
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:database/cart_items/delete-all-cart-items.sql",
```
```suggestion
            "classpath:database/cart/items/delete-all-cart-items.sql",
```
you can google standards of naming packages in java projects. If I'm not mistaken it is recommended to have 1 word for a package, but not `_`

yevheniisynytsia commented on src/test/java/online/book/store/repository/ShoppingCartRepositoryTest.java:
```diff
@@ -0,0 +1,53 @@
+class ShoppingCartRepositoryTest {
+    private static User user;
+    private static ShoppingCart shoppingCart;
+    private static final String CANT_FIND_BOOK_BY_ID_MSG = "Can't find book by id: ";
```
unused

yevheniisynytsia commented on src/test/java/online/book/store/repository/ShoppingCartRepositoryTest.java:
```diff
@@ -0,0 +1,53 @@
+    @BeforeAll
+    static void beforeAll() {
+        user = new User()
+                .setId(2L)
```
I would recommend moving id to constant and reuse it in `findByUserId` so that we can't misclick and look for user with id 3 and so on

yevheniisynytsia commented on src/test/java/online/book/store/service/imp/ShoppingCartServiceImplTest.java:
```diff
@@ -0,0 +1,76 @@
+        Mockito.when(userRepository.findById(1L)).thenReturn(Optional.of(user));
+        Mockito.when(shoppingCartRepository.findByUserId(1L)).thenReturn(Optional.of(shoppingCart));
+        Mockito.when(shoppingCartMapper.toResponseDto(shoppingCart))
+                .thenReturn(shoppingCartResponseDto);
```
```suggestion
                .thenReturn(shoppingCartResponseDto);
                
```

yevheniisynytsia reviewed (changes requested):
Well done! Some style comments left

DEz-1996 committed:
Some fixes:
1. Refactor some directories
2. Create constant Long userId in ShoppingCartRepositoryTest
3. Create constant Long userId in ShoppingCartServiceImplTest

DEz-1996 committed:
Some fixes:
1. Refactor some directories
2. Create constant Long userId in ShoppingCartRepositoryTest
3. Create constant Long userId in ShoppingCartServiceImplTest

yevheniisynytsia reviewed (approved):
:fire:
